aten/src/ATen/core/ivalue.h

```diff
@@ -91,7 +91,39 @@ using DoubleList = List<double>;
 using BoolList = List<bool>;
 using GenericList = List<IValue>;
 
+// User-defined object.
+template <typename Elem>
+struct CAFFE2_API UserObjectGeneric final : c10::intrusive_ptr_target {
+ public:
+  UserObjectGeneric(std::string name, size_t numSlots)
```

Comment:
Can't we use interned strings for the name? Right now every instance of an object will keep a separately allocated `std::string`. Seems really wasteful

Reply:
Or at least make this a `shared_ptr`

Reply:
good catch, will change